test/testutils.jl

```diff
@@ -1,23 +1,44 @@
-allapprox(coords₁::C, coords₂::C; kwargs...) where {C<:CRS} =
-  all(ntuple(i -> isapprox(getfield(coords₁, i), getfield(coords₂, i); kwargs...), nfields(coords₁)))
-
-function allapprox(coords₁::C, coords₂::C; kwargs...) where {C<:Cartesian2D}
-  isapprox(coords₁.x, coords₂.x, kwargs...) && isapprox(coords₁.y, coords₂.y, kwargs...)
+svec(coords::Cartesian) = SVector(getfield(coords, :coords))
+svec(coords::CoordRefSystems.Projected) = SVector(coords.x, coords.y)
+
+is_approx_angle(
+  α, β;
+  atol = sqrt(eps(2 * convert(promote_type(Unitful.numtype.((α, β))...), π))),
+) = isapprox(mod2pi(α - β + π), π; atol = ustrip(rad, atol))
+
+allapprox(coords₁::Cartesian{Datum}, coords₂::Cartesian{Datum}) where {Datum} = svec(coords₁) ≈ svec(coords₂)
+allapprox(coords₁::Polar{Datum}, coords₂::Polar{Datum}) where {Datum} = coords₁.ρ ≈ coords₂.ρ && is_approx_angle(coords₁.ϕ, coords₂.ϕ)
+allapprox(coords₁::Cylindrical{Datum}, coords₂::Cylindrical{Datum}) where {Datum} = SVector(coords₁.ρ, coords₁.z) ≈ SVector(coords₁.ρ, coords₂.z) && is_approx_angle(coords₁.ϕ, coords₂.ϕ)
+allapprox(coords₁::Spherical{Datum}, coords₂::Spherical{Datum}) where {Datum} = coords₁.r ≈ coords₂.r && is_approx_angle(coords₁.θ, coords₂.θ) && is_approx_angle(coords₁.ϕ, coords₂.ϕ)
```

Comment:
I believe we can omit the `Datum` in all these definitions given that it is not used.

Reply:
We do need to parametrise on `Datum` to exclude cases like 
```julia
allapprox(
  Cartesian{Datum1}(...), 
  Cartesian{Datum2}(...),
)
```
Cases like these wouldn't be handled correctly by the methods here. 

This same reasoning applies also to the latlon and projected cases below.